### PR TITLE
Test exporting batch norm, dropout, RNN

### PR DESCRIPTION
**Pull request by jamesr66a**

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -1441,6 +1441,57 @@ def constant_prop():
...
+        imported = self.getExportImportCopy(traced)
+        x = torch.randn(3, 4)
+        self.assertEqual(traced(x), imported(x))
+
+    def test_export_batchnorm(self):
+        for mode in ['eval', 'train']:
+            class BatchnormTest(torch.nn.Module):
+                def __init__(self):
+                    super(BatchnormTest, self).__init__()
+                    self.batchnorm = torch.nn.BatchNorm1d(100)
```
Comment (soumith) [inline test/test_jit.py]: also test `torch.nn.BatchNorm1d(100, affine=False)`

Also test `torch.nn.BatchNorm2d(100)` and `torch.nn.BatchNorm2d(100, affine=False)`

File: test/test_jit.py [hunk 2]
```diff
@@ -1583,6 +1583,93 @@ def foo(x):
...
+        class DropoutTest(torch.nn.Module):
+            def __init__(self):
+                super(DropoutTest, self).__init__()
+                self.dropout = torch.nn.Dropout()
+
+            def forward(self, x):
+                return self.dropout(x)
+
+        test = DropoutTest()
+        test.eval()
```
Comment (apaszke) [inline test/test_jit.py]: Testing dropout export in eval mode doesn't seem to be the best idea, because an identity graph would pass this test too. Can't we use `torch.random.fork_rng`?

File: test/test_jit.py [hunk 3]
```diff
@@ -1583,6 +1583,93 @@ def foo(x):
...
+    def test_export_dropout(self):
+        class DropoutTest(torch.nn.Module):
+            def __init__(self):
+                super(DropoutTest, self).__init__()
+                self.dropout = torch.nn.Dropout()
+
+            def forward(self, x):
+                return self.dropout(x)
+
+        test = DropoutTest()
```
Comment (apaszke) [inline test/test_jit.py]: I'm pretty sure this module is 100% equivalent to just using `torch.nn.Dropout` directly.

**Review by apaszke — APPROVED** [review]

File: test/test_jit.py (hunk 2 above)
Comment (zdevito) [reply test/test_jit.py]: I am also just ok keep the test as is. In some sense, I just wanted to make sure networks in eval mode with dropout would export correctly as a smoke test that we didn't exclude a ton of models. This was because previously Dropout was a special python operator.
